lib/nokogiri/xml/syntax_error.rb

```diff
@@ -19,14 +19,23 @@ def aggregate(errors)
         end
       end
 
+      # What part of libxml2 raised this error (enum xmlErrorDomain)
       attr_reader :domain
+      # libxml2 error code (enum xmlParserErrors)
       attr_reader :code
+      # libxml2 error level (enum xmlErrorLevel)
       attr_reader :level
       attr_reader :file
       attr_reader :line
+      # libxml2 path of the node in the tree that caused the error
+      attr_reader :path
+      # libxml2 extra string information
       attr_reader :str1
+      # libxml2 extra string information
       attr_reader :str2
+      # libxml2 extra string information
       attr_reader :str3
+      # libxml2 extra extra number information
```

Comment:
These comments, though probably an improvement over nothing, still don't feel very informative. Over time I'm trying to evolve Nokogiri towards being less-libxml2-specific and I'm not sure which of these attributes would make that cut.

Let's keep the `:path` one since we're adding it, but let's not tie it to libxml2 in the description. And I think the rest of these comments can be omitted in the name of making it easier for me to `:nodoc:` them later if I decide to.

Reply:
This makes sense, thanks for all your time mike.

As for which ones that make the cut, I would probably say code and domain don't nor all the str#

I imagine the ideal API instead of returning code + str# + int# it would be a new error object with the values per code. This would be relevant to all libraries. The mapping would be annoying but it would be a great user experience.

Will make changes now